internal/statestore/public.go

```diff
@@ -59,5 +60,11 @@ type Service interface {
 
 // New creates a Service based on the configuration.
 func New(cfg config.View) Service {
-	return newRedis(cfg)
+	s := newRedis(cfg)
+	if cfg.GetBool(monitoring.ConfigNameEnableMetrics) {
+		return &instrumentedService{
```

Comment:
Rather than creating an instrumented service, can we rather have a shim on the record that checks for the enabled setting to decide whether to record or not?

Reply:
I wanted something that when/if we move to another backend we can use the same wrapper instead of re-adding the logic there.

Reply:
Ah. I misunderstood. Do we need to config control it at all then? Why not just have public implement the wrapper itself - so you always return the wrapped implementation?